express.js

```diff
@@ -318,36 +320,76 @@ app.get("/tracks/:id/:time", cors(), async (req, res) => {
 
 app.post("/imageUpload", upload.single("file"), async (req, res) => {
   //Help from https://stackoverflow.com/questions/36477145/how-to-upload-image-file-and-display-using-express-nodejs
-  let file = __dirname + "\\tmp\\" + req.file.originalname;
-  fs.rename(req.file.path, file, function (err) {
-    if (err) {
-      console.log(err);
-      res.send(500);
-    } else {
-      gm()
-        .command("convert")
-        .in(file)
-        .define("jpeg:extent=256kb")
-        .stream(function (err, stdout, stderr) {
-          if (err) {
-            console.log("ERROR: ", err);
-          } else {
-            let writeStream = fs.createWriteStream(
-              __dirname + "\\tmp\\playlistImg.jpg"
-            );
-            writeStream.on("error", function (e) {
-              console.log("ERR: ", e);
-            });
-            stdout.pipe(writeStream);
-            writeStream.on("finish", function () {
-              res.sendFile(__dirname + "\\tmp\\playlistImg.jpg");
+  req.params.id = 'spotify:dragonguy128';
+  let expDate = Date.parse(
+    await redisClient.hgetAsync(`${req.params.id}`, "expiresAt")
+  );
+  let curDate = new Date();
+  if (curDate > expDate) {
+    refreshSpotifyToken(req.params.id);
+  }
+  let accessToken = await redisClient.hgetAsync(
+    `${req.params.id}`,
+    "accesstoken"
+  );
+  if (accessToken) {
+    spotifyApi.setAccessToken(accessToken);
+    try {
+      var file = __dirname + "\\tmp\\" + req.file.originalname;
+      fs.rename(req.file.path, file, function (err) {
+        if (err) {
+          console.log(err);
+          res.send(500);
+        } else {
+          gm()
+            .command("convert")
+            .in(file)
+            .define("jpeg:extent=256kb")
+            .stream(function (err, stdout, stderr) {
+              if (err) {
+                console.log("ERROR: ", err);
+              } else {
+                let writeStream = fs.createWriteStream(
+                  __dirname + "\\tmp\\playlistImg.jpg"
+                );
+                writeStream.on("error", function (e) {
+                  console.log("ERR: ", e);
+                });
+                stdout.pipe(writeStream);
+                writeStream.on("finish", async function () {
+                  let send = Buffer.from(__dirname + "\\tmp\\playlistImg.jpg", 'binary').toString('base64');
```

Comment:
I think buffer.from is interpreting this as a String, not getting the file and encoding that. According to this, we should be able to have the writeStream point to a Buffer instead and encode and send that directly, without writing the file to disk as a middle step. Max file size is 256kb, so it shouldnt be too hard on memory https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options